devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/extension/QuarkusPluginExtension.java

```diff
@@ -215,12 +216,16 @@ public Path appJarOrClasses() {
         return classesDir;
     }
 
-    public Map<String, String> getQuarkusBuildProperties() {
+    public MapProperty<String, String> getQuarkusBuildProperties() {
```

Comment:
Yes, it has been released already, unfortunately.

Reply:
@aloubyansky even if this is a breaking change, who would it affect?

Reply:
@jacobdotcosta would this change break your use-case?

Reply:
This change can break builds depending on how people use `QuarkusPluginExtension`:

```kotlin
// works
quarkus {
  set("prop", "some value")
}

quarkus {
  quarkusBuildProperties.put("prop", "some prop")
}

// assignment works, passing the reference may be broken
val props = quarkus.quarkusBuildProperties

// broken
val props: Map<String, String> = quarkus.quarkusBuildProperties

// broken
quarkus.quarkusBuildProperties.entrySet()
```

So if you want to go the safe route, it would be better to add a new accessor method instead, potentially deprecating the old one.

Reply:
Given that it's a very recent feature and this change encourages a good practice, I'd accept this change.